src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -31,7 +35,32 @@
     private static final boolean DEBUG_STATS = false;
 
     /**
-     * Loads the alignment regions and sequence of all locally-assembled contigs from the text file they are in;
+     * Converts a {@link GATKRead}, particularly the alignment and sequence information in it to the custom {@link AlignmentRegion} format.
+     */
+    @VisibleForTesting
+    static Tuple2<Iterable<AlignmentRegion>, byte[]> convertToAlignmentRegions(final Iterable<GATKRead> reads) {
+        final List<GATKRead> gatkReads = IterableUtils.toList(reads);
+        final Optional<GATKRead> primaryRead = gatkReads.stream().filter(r -> !(r.isSecondaryAlignment() || r.isSupplementaryAlignment())).findFirst();
+        if (! primaryRead.isPresent()) {
+            throw new GATKException("no primary alignment for read " + gatkReads.get(0).getName());
+        }
+
+        final byte[] bases = primaryRead.get().getBases();
+        if (primaryRead.get().isReverseStrand()) {
+            SequenceUtil.reverseComplement(bases, 0, bases.length);
+        }
+
+        final Iterable<AlignmentRegion> alignmentRegionIterable = IteratorUtils.asIterable(gatkReads.stream()
```

Comment:
This will be a single-use Iterable (i.e., one that can only iterate once, and is always empty thereafter).  That's probably OK, but it's certainly not clear that it will be without examining the calling code.